File: gestion_note/app/Models/UE.php
<?php

namespace App\Models;

use App\Models\Matiere;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class UE extends Model
{
    use HasFactory;

    protected $table = "ue";

    protected $primaryKey = 'Id_UE';
    protected $fillable = ['Designation', 'Credit', 'Niveau', 'Parcours'];

    public function matiere()
    {
        return $this->hasMany(Matiere::class);
    }
}

File: myapp/lib/Models/UE.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

import 'package:myapp/pages/EtudiantViews/listEtudiants.dart';

class UE{
  int Id_UE;
  String Designation="";
  int Credit;
  String Niveau="";
  String Parcours="";


  UE(this.Id_UE, this.Designation, this.Credit, this.Niveau,this.Parcours);

  Map toJson() => {
    'Id_UE':Id_UE,
    'Designation':Designation,
    'Credit':Credit,
    'Niveau':Niveau,
    'Parcours':Parcours
  };
}